autode/opt/optimisers/dimer.py

```diff
@@ -0,0 +1,358 @@
+            return _StepResult.skipped_rotation
+
+        self._rotate_coords(phi_1, update_g1=True)
+
+        b1 = 0.5 * dc_dphi0                              # eqn. 8 from ref. [1]
+
+        a1 = ((c_phi0 - self._c + b1 * np.sin(2*phi_1))  # eqn. 9 from ref. [1]
+              / (1 - 2.0 * np.cos(2.0 * phi_1)))
+
+        a0 = 2.0 * (c_phi0 - a1)                        # eqn. 10 from ref. [1]
+        phi_min = Angle(0.5 * np.arctan(b1 / a1), units='radians')
+        logger.info(f'ϕ_min = {phi_min.to("degrees"):.4f}º')
+
+        c_min = 0.5 * a0 + a1 * np.cos(2.0*phi_min) + b1 * np.sin(2.0*phi_min)
+
+        if c_min > c_phi0:
+            logger.info('Optimised curvature was larger than the initial, '
+                        'adding π/2')
+            phi_min += np.pi / 2.0
+
+        # Rotate back from the test point, then to the minimum
+        self._coords = cached_coordinates
+        self._rotate_coords(phi=phi_min, update_g1=True)
+
+        return _StepResult.did_rotation
+
+    def _translate(self, update_g0=True) -> '_StepResult':
+        """Translate the dimer under the translational force"""
+        x0 = self._coords.x0
+
+        trns_iters = [c for c in self._history if c.did_translation]
+
+        if len(trns_iters) < 2:
+            step_size = float(self.init_alpha)
+            logger.info(f'Did not have two previous translation step, guessing'
+                        f' α = {step_size} Å')
+
+        else:
+            prev_trns_iter = trns_iters[-2]
+            logger.info(f'Did {len(trns_iters)} previous translations, can '
+                        f'calculate α using the Barzilai–Borwein method')
+
+            step_size = (np.abs(np.dot((x0 - prev_trns_iter.x0),
+                                       (self._coords.f_t - prev_trns_iter.f_t)))
+                         / np.linalg.norm(self._coords.f_t - prev_trns_iter.f_t)**2)
+
+        delta_x = step_size * self._coords.f_t
+        trns_rms = MWDistance(np.sqrt(np.mean(np.square(delta_x))))
+
+        if trns_rms < self.trns_tol:
+            logger.info(f'Step length small than tolerance {self.trns_tol}')
+            return _StepResult.skipped_translation
+
+        logger.info(f'Translating by ~{trns_rms:.4f} per coordinate')
+
+        coords = self._coords.copy()
+        coords += delta_x
+
+        self._coords = coords
+        self._coords.phi = Angle(0.0)    # Did not rotation
+        self._coords.dist = trns_rms     # but did translate
+
+        if update_g0:
+            self._update_gradient_at(DimerPoint.midpoint)
+
+        return _StepResult.did_translation
+
+    def _initialise_run(self) -> None:
+        """Initialise running the dimer optimisation"""
+
+        if np.isclose(self._coords.delta, 0.0):
+            raise RuntimeError('Zero distance between the dimer points')
+
+        self._coords._g = np.zeros(shape=(3, 3*self._species.n_atoms))
+
+        # TODO: Hessian
```

Comment:
hmmm

Reply:
Will improve this comment... no time to update sadly